packages/assets-controllers/src/TokensController.ts

```diff
@@ -506,9 +506,8 @@ export class TokensController extends BaseController<
 
     // Used later to dedupe imported tokens
     const newTokensMap = [
-      ...(allTokens[interactingChainId ?? this.#chainId]?.[
-        this.#getSelectedAccount().address
-      ] || []),
+      ...(allTokens[interactingChainId]?.[this.#getSelectedAccount().address] ||
```

Comment:
OK so we are removing the `this.#chainId` check as this could be stale/incorrect?